Refractor/simplify with column renderer

scoufal opened the pull request:
Unifikace zobrazení tabulek, kromě finančních.

Používá defaultní předefinovatelné IColumnHeaderRenderer a IColumnContentRenderer
Pro oddělování v tabulkách slouží IRowTextPainter a IBreakRowDetector

Část tabulek je refraktorována, zvláště ty které obsahují Race Capacity Limit.

Tabulky se skrytými roky nepoužívalí class jako filter ale data-group attribut.


**PR 28 zahrnuje i PR 27 feature Race Capacity limit**

scoufal committed:
Refractor Races table rendering

- general interfaces for header&column renderer, row break and row painter
- general implemetation for generic field names or independent
- special implementation for races table

Open
- financial tables with hidden rows and button activation
- only two tables extended by count of participants/capacity

scoufal committed:
Improve show/hide years in races table

- implement row class/attribute extension
- remove class extension missuse for year visualisation control
- use data-group attribute for row year grouping

scoufal committed:
Rework msn_races

- last races table

scoufal committed:
Unify races list

- generealize count of registered to be easy placed where required
- remove obsolete files with races list

scoufal committed:
Extend abstraction for mutliple table types

- abstract base class for renderer factory
- generalize TableColumn and RendererTable
- move specific functions into derived factory renderer class

scoufal committed:
Extend renderer for races support

- replace some race tables with renderer
- add row information number, count
- add optional table row filter
- remove unused ex_td from ctable
- add onclick option to ctable header

scoufal committed:
Merge branch 'arnost00:master' into Refractor/Simplify_with_column_renderer

scoufal committed:
Merge branch 'Feature/limit_na_ubytovani' into Refractor/Simplify_with_column_renderer

arnost00 commented on _terminy_rg.php:
Duvod zruseni ?

scoufal replied:
Soubor bez odkazu. Pokud ho někdo píše do URL ručně, tak se ozve.

Struktura souboru neodpovídá ostatním a nevěděl jsem jak ho udržovat.

scoufal replied:
Ten začátek je aspoň 12 let starý : ( , doporučil bych přesunutí do death branch

xkenia replied:
Nepouzivam a souhlasim, ze je tvuj, muj by nebyl tak pekne nastylovany :-D

arnost00 committed:
Merge branch 'master' into Refractor/Simplify_with_column_renderer

arnost00 commented:
Jeste jsem si vsiml ze sloupec "Účast" (Přihlášeno/Kapacita) se vyskytuje pouze v prihlaskach jednotlivce, nemel by se vyskytovat i v tabulkach pro prihlasky Trenera a Prihlasovatele ?
A to ani nemluvim ze v prihlaskach po jednom pro Trenera a Prihlasovatele se take zadna kapacita nezobrazuje ... to ma nejaky duvod ze je to jen v zakladnich prihlaskach ?

arnost00 commented:
Jinak se zatim zda ze se to chova ocekavane. Nove tridy na tabulky snad casem take prostuduji.

scoufal commented:
PR #27 a PR#28 jsou podmíněné historicky.

PR #27 by požadavek od organizátora limitovaných tréningů. Implementovaný v minimlizované variantě. Pro integraci do dalších tabulek by změny musely být kopírovány a tak jsem vytvořil PR #28 , abych to nemusel dělat pořád dokola. Jenže nebylo jasné jestli má naději na přijmutí, když je to koncepcionální změna.

scoufal committed:
In all "Přihlášky na závody" show Limit column and make old races list expandable

- collect capacity data and show capacity column if configured
- add year expander break if old races filter is active
- unify data load from SQL response

scoufal committed:
Merge branch 'Refractor/Simplify_with_column_renderer' of https://github.com/scoufal/members into Refractor/Simplify_with_column_renderer

scoufal committed:
Change year expander style from button to ▼ 2025

- there might be still the old button active, the js is compatible with boths
- the backgroud color is defined by even/odd row

scoufal committed:
Unify race forms for limit and table renderer

- change text of title with limit
- add limit information to additional forms
- use table rendering in forms without limit visualisation

This change does not affect bulk registration forms

arnost00 commented:
> To je tak, když se to zkouší na malé sadě dat :)
> 
> Ten SQL dotaz počítá přihlášené účastníky v závodech, což při celé historii je docela hodně závodů. Výkonově by bylo asi nejlepší to někde materializovat ve view nebo tabulce, nebo propisovat na definici závodu.
> 
> Jednodušší varianta je počítat jen pro závody max 1 rok zpět a/nebo s nastavenou kapacitou.
> 
> Co vám připadá lepší?

Ja bych asi hlasoval pro view, ale vetsi specialista na SQL je @xkenia. Tak bych rad kdyby se k tomu vyjadril.

scoufal committed:
Use persistent field for count of participants

- add new field count of participants for race
- update current value by actual count
- create trigger for include/delete rows from participants table
- us the  participant field instead of on the fly participant count

arnost00 commented:
Hmmm ja za view povazuju vec co jsme pouzivali uz cca kolem roku 2008, a dodnes to v MariaDb jde vytvorit.
Prikladam priklad SQL dotazu pro vytvoreni view z cca roku 2008 : ```CREATE OR REPLACE VIEW `eobcz_zbm`.`v_main_payments` AS select `m`.`nazev` AS `nazev`,`u`.`zavod` AS `zavod`,sum(`u`.`vklad`) AS `vklad_za_prihlasene` from (`eobcz_zbm`.`a_main` `m` left join `eobcz_zbm`.`a_users` `u` on((`m`.`zkratka` = substr(`u`.`reg`,1,3)))) where (`u`.`id` is not null) group by `m`.`zkratka`,`u`.`zavod` order by `m`.`zkratka`;```
Bylo to v nasem pokusu o prihlaskovy system na zavody (nez se prosadila OBHana :)

Co k tomu rekne @xkenia ?

scoufal commented:
Některé DB materialized views mají. MySQL a MariaDB ne. VIEW z MariaDB nemá cenu, ta se provádí vždy znovu.

V MariaDB se používají triggery, aspoň podle ChatGPT

scoufal committed:
Count participans by PHP instead of SQL triggers

arnost00 commented on common_race.inc.php:
```diff
@@ -503,30 +503,4 @@ function toggleCategoriesAndScroll() {
-
-	}
-	return $count_registered;
-}
```
ok
